lib/common_test/doc/src/ct_hooks_chapter.xml

```diff
@@ -454,19 +454,21 @@
 
  %% Called before each init_per_testcase.
  pre_init_per_testcase(Suite,TC,Config,State) ->
-     {Config, State#state{ ts = now(), total = State#state.suite_total + 1 } }.
+     Now = erlang:monotonic_time(microsecond),
```

Comment:
Mind you where this once returned a [`timestamp()`](https://www.erlang.org/doc/man/erlang.html#type-timestamp), it's now returning an integer. Maybe a better equivalent to `now()` would be `erlang:timestamp()`?

Reply:
Yeah, perhaps.

`erlang:timestamp/0` is probably just fine if it makes the example simpler. And the monotonic part is probably not very interesting when analyzing test suite run times. It would make the diff in this PR smaller since it can be used with the same `now_diff`.

I just followed the "Measure Elapsed Time" advice in the docs from the deprecation and arrived at the solution in the PR:

> ##### Measure Elapsed Time
>
> Take time stamps with [erlang:monotonic_time/0](https://www.erlang.org/doc/man/erlang.html#monotonic_time-0) and calculate the time difference using ordinary subtraction. The result is in native [time unit](https://www.erlang.org/doc/man/erlang.html#type_time_unit). If you want to convert the result to another time unit, you can use [erlang:convert_time_unit/3](https://www.erlang.org/doc/man/erlang.html#convert_time_unit-3).
>
> An easier way to do this is to use [erlang:monotonic_time/1](https://www.erlang.org/doc/man/erlang.html#monotonic_time-1) with the desired time unit. However, you can then lose accuracy and precision.

Reply:
Sure... In the same documentation, though, you'll find

> **Don't**
Use erlang:now/0 to retrieve the current Erlang system time.

> **Do**
Use [erlang:system_time/1](https://www.erlang.org/doc/man/erlang.html#system_time-1) to retrieve the current Erlang system time on the [time unit](https://www.erlang.org/doc/man/erlang.html#type_time_unit) of your choice.
If you want the same format as returned by erlang:now/0, use [erlang:timestamp/0](https://www.erlang.org/doc/man/erlang.html#timestamp-0).

I'm pretty sure backward compatibility isn't important here, but I was suggesting the change to have a simpler example. 😄 

And upon closer inspection I missed `timer:now_diff(now(), State#state.ts)` from your changes, so it's probably Ok to keep as-is.

Reply:
Oh, I see what you mean then. Yeah I changed both sides of the subtraction.

I agree, it would look like a simpler change if we use `erlang:timestamp()` instead. I don't have any strong opinion, I just followed the deprecation link and scanned the headings to find "Measure Elapsed Time" and thought that this example code was about measuring elapsed time more than "Retrieve Erlang System Time" so I didn't see the timestamp suggestion.

Reply:
Hmm, the link from the reference manual for CT to this example section in the user guide says:

> For a minimal example of a CTH, see section [Example CTH](https://www.erlang.org/doc/apps/common_test/ct_hooks_chapter.html#example) in the User's Guide.

And I don't think this example is minimal.

Here is a version where only the warnings have been removed:
https://gist.github.com/fabjan/cc178d21f42a773cd45e43fe1b1416d7

Here is a minimal version keeping all the features of the current example: https://gist.github.com/fabjan/754840a6daf30058c00880b465448974

Reply:
But I also tried the code now for real and it's not really working at all any way I do it on my local machine, regardless of any changes.

The hooks crash when trying to use the opened file:
```
*** CT 2022-11-26 16:28:33.664 *** Suite Hook
Call to CTH failed: error:{terminated,
                           [{io,format,
                             [<0.656.0>,"~p.~n",
                              [{test_run,1,
                                [{suites,foo_SUITE,0,
                                  [{testcase,foo_SUITE,test,ok,
                                    409}]}]}]],
                             [{file,"io.erl"},
                              {line,94},
                              {error_info,
                               #{cause => {io,terminated},
                                 module => erl_stdlib_errors}}]},
                            {example_cth,terminate,1,
                             [{file,
                               "/Users/fabian/code/example_cth/src/example_cth.erl"},
                              {line,58}]},
                            {ct_hooks,catch_apply,3,
                             [{file,"ct_hooks.erl"},{line,491}]},
                            {ct_hooks,call_terminate,3,
                             [{file,"ct_hooks.erl"},{line,218}]},
                            {ct_hooks,call,4,
                             [{file,"ct_hooks.erl"},{line,301}]},
                            {ct_hooks,terminate,1,
                             [{file,"ct_hooks.erl"},{line,100}]},
                            {ct_hooks,terminate_if_scope_ends,3,
                             [{file,"ct_hooks.erl"},{line,374}]},
                            {ct_hooks,call,4,
                             [{file,"ct_hooks.erl"},{line,306}]}]}
```

It works If I move the file open to the terminate function together with the writing.
But if I use the hook in multiple suites (started via `Module:suite/0`) they stomp over each other's content in the log file.

Is this a problem with my local machine perhaps? Or is this related to the information at https://www.erlang.org/doc/apps/common_test/ct_hooks_chapter.html#cth-processes-and-tables

> CTHs are run with the same process scoping as normal test suites, that is, a different process executes the init_per_suite hooks then the init_per_group or per_testcase hooks. So if you want to spawn a process in the CTH, you cannot link with the CTH process, as it exits after the post hook ends. Also, if you for some reason need an ETS table with your CTH, you must spawn a process that handles it.

Is the IoDevice returned by file:open a process that is linked?

The fact that the suite data is overwritten by the next suite might be due to how I installed the hook I guess.

I'm creating a separate little repository where I can run some real code and tests, instead of just manipulating this example code inside a documentation article and running it in my head: https://github.com/fabjan/example_cth

(I don't expect anyone to treat this PR any differently, at this point I'm just curious as to how one could make this CT hook actually work)